---
title: "Calculate PRS"
author: X Shen  
date: "\n`r format(Sys.time(), '%d %B, %Y')`"
output: github_document
---

For more information about PRSice 2.0, see its [wiki page](http://prsice.info/). For GCTA, see [wiki page](https://cnsgenomics.com/software/gcta).

------------------------------------------------------------------------

## Set up a local copy of the github repository

Clone files:

```{bash, eval=F}
git clone https://github.com/xshen796/ENIGMA_mdd_prs.git
cd ENIGMA_mdd_prs
```

Set permission to files:

```{bash, eval=F}
find . -executable -exec chmod +x {} \;
```

------------------------------------------------------------------------

## Environment preparation

R libraries required:

-   dplyr
-   data.table
-   optparse

Open R, test if these packages are installed.

```{r, eval = F}
library(dplyr)
library(optparse)
library(data.table)
```

If R gives an error message, install the package(s) using:

```{r, eval = F}
install.packages('dplyr')
install.packages('optparse')
install.packages('data.table')
```

------------------------------------------------------------------------

## File preparation

#### Plink-format genetic data

At least three files should be included with suffixes: .bim, .bed, .fam. Please make sure genetic data is under genome build hg19/GRCh37.

If you are not sure, simply run the PRS pipeline and it will check automatically.

#### Phenotype file

Create a dummy file with the .fam file. Use command given below.

```{bash, eval = F}
Rscript util/create_dummypheno.R --fam <.bim file location>
```

Here is an example using plink data in the *data* folder:

```{bash, eval = F}
Rscript util/create_dummypheno.R --fam data/1000G_eur_chr22.fam
```

A file will be created as *data/dummy_pheno.txt .*

#### 1000genome plink files

If your sample has less than 1000 participants, it is better to use the 1000genome data as LD reference.

You can download the plink files and store in the *data* folder. Use the command below. You'll be asked to give a password for each file. Password: enigma.

```{bash, eval=F}
cd data

curl -u "NvBAVLFOZP7o6kC" "https://datasync.ed.ac.uk/public.php/webdav" -o 1000g_CEU_plink.fam
curl -u "yk7HILluLWfUdjQ" "https://datasync.ed.ac.uk/public.php/webdav" -o 1000g_CEU_plink.bim
curl -u "QWagF7h7qJcOK3D" "https://datasync.ed.ac.uk/public.php/webdav" -o 1000g_CEU_plink.bed

cd ../
```

!!! The PRS script will detect if your sample is too small (N\<1000).

------------------------------------------------------------------------

### Run PRSice

Run the job file to generate all PRS using the command below.

```{bash, eval=F}
bash job.PRS_prsice2.sh -d <base name for plink-format files>
```

Here is an example using plink data in the *data* folder:

```{bash, eval = F}
bash job.PRS_prsice2.sh -d data/1000G_eur_chr22
```

**!!** If you see an error about duplicated variants looking like below, run the above command again. The script will pick up the valid SNP list when re-calculating PRS.

> Error: A total of 7456 duplicated SNP ID detected out of 5958288 input SNPs! ...

------------------------------------------------------------------------

### A checklist for output files

Check if all files present in the *PRS* folder. A checklist can be found [here](https://github.com/xshen796/ENIGMA_mdd_prs/blob/main/script/PREP_PRS/CheckList_output.md).

------------------------------------------------------------------------

### Upload PRS and supportive files

Upload the entire *PRS* folder to the medflix server. See a step-by-step guide [here](https://github.com/xshen796/ENIGMA_mdd_prs/blob/main/docs/Accessing%20your%20folder%20on%20MediaFlux%20ENIGMA%20MDD%20storage%20system%20updated%20Aug2020%5B2%5D.pdf).
